src/tools/debugger/session/Breakpoints.java

```diff
@@ -222,4 +236,59 @@ public static AbstractBreakpointNode createOpposite(final SourceSection source)
       return new DisabledBreakpointNode();
     }
   }
+
+  public void setActorStepping(final StepActorOperation step) {
+    this.stepping = new Stepping(step);
+  }
+
+  public Stepping getStepping() {
+    return this.stepping;
+  }
+
+  /**
+   * Return the stepping operation corresponding for a source section.
+   */
+  public static SteppingType checkSteppingOperation(final SourceSection source) {
+    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+      FullSourceCoordinate sourceCoord = SourceCoordinate.create(source);
+      Breakpoints breakpointCatalog = VM.getWebDebugger().getBreakpoints();
+      if (breakpointCatalog.getStepping() != null) {
+        return breakpointCatalog.getStepping().getSteppingTypeOperation(sourceCoord);
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Disable existing breakpoints and runs execution before the next root node is executed.
+   */
+  public void prepareStepIntoMessage() {
+    disableBreakpoints();
```

Comment:
why do you disable breakpoints? I can't imagine that this is what a user wants.
If I am stepping, and there are breakpoints sets, why wouldn't I want them to trigger?